example/createCounter.js

```diff
@@ -39,16 +41,35 @@ export default function (rootElem: HTMLElement): Counter {
 		return _value;
 	}
 
+	let _fakeSuccessValue = false;
+	let _fakeErrorValue = false;
+
 	clientOpened.then(() => {
 		// simple regression steps
 		client.publishStep('.* call a step with a result', () => 'hello!');
 		client.publishStep('.* call a step without a result', () => {});
 		client.publishStep('.* call a step which fails', () => { expect(true).toBe(false); });
+		client.publishStep('.* call a step which succeeds asynchronously', () => {
```

Comment:
Mechanism looks right. I wonder if we'll be able to find a more elegant way to express the retries at some point later - most likely this is possible with client code already given the present design  e.g. 
client.publishStep("my step", () => {}, withRetry(1)")

Generally a user would want to specify the retry period in seconds I think - rather than millis.
To match what we have in Java the ability to specify an alternative unit would occasionally be useful
e.g. withRetry(milliseconds(1000)) or similar